server/src/apub/community_inbox.rs

```diff
@@ -42,28 +33,36 @@ pub async fn community_inbox(
 fn handle_follow(
   follow: &Follow,
   request: &HttpRequest,
-  conn: &PgConnection,
+  community_name: &str,
+  db: DbPoolParam,
+  _chat_server: ChatServerParam,
 ) -> Result<HttpResponse, Error> {
   let user_uri = follow
     .follow_props
     .get_actor_xsd_any_uri()
     .unwrap()
     .to_string();
-  let user = fetch_remote_user(&Url::parse(&user_uri)?, conn)?;
-  verify(&request, &user.public_key.unwrap())?;
-
-  // TODO: make sure this is a local community
-  let community_uri = follow
+  let _community_uri = follow
     .follow_props
     .get_object_xsd_any_uri()
     .unwrap()
     .to_string();
-  let community = Community::read_from_actor_id(conn, &community_uri)?;
+
+  let conn = db.get()?;
+
+  let user = get_or_fetch_and_upsert_remote_user(&user_uri, &conn)?;
+  let community = Community::read_from_name(&conn, &community_name)?;
+
+  verify(&request, &user.public_key.unwrap())?;
+
   let community_follower_form = CommunityFollowerForm {
     community_id: community.id,
     user_id: user.id,
   };
+
+  // This will fail if they're already a follower
```

Comment:
Right, I think it would be a good idea to ignore the error in this case and send an Accept back anyway.

Reply:
Dang all of these things are difficult to think about, but I think you're right. I'll ignore the result.